### PR TITLE
E2E tests

### PR DESCRIPTION
**Pull request by JMach1**
Add infra for running e2e tests. 

- Mock server per each test => flexibility for different cases
- pipeline
- examples of e2e tests

### CODE REVIEW

File: package.json
Comment (harunrst) [inline package.json:21]: @f-alizada you might want to sync up with @JMach1 
he is running a specific test folder for e2e. And your branch is running all tests, we are double running tests.

Comment (f-alizada) [reply package.json:21]: This PR is on hold currently, and from change we can see that for e2e tests we have separate folder so both pr's running different tests

Comment (harunrst) [reply package.json:21]: hmm yes, it is under different folder than src, okay it is cool here then

**Review by DeagleGross — APPROVED** [review]
LGTM, but suggesting to wait for devPortal expert to review

File: .github/workflows/test.yml
Comment (Mielek) [inline .github/workflows/test.yml:36]: is the & at the end needed?

**Review by malincrist — APPROVED** [review]

File: tests/e2e/runtime/apis.spec.ts
Comment (Mielek) [inline tests/e2e/runtime/apis.spec.ts:37]: I would suggest as you have a great opportunity to create a great example of test. In my opinion two elements are missing.

1.  Abstraction to set backend data - Abstraction which would make creating data easy and test very readable.
2.  Object oriented representation of page and page elements - This is for future. As tests will get bigger and more complex parts will be reused in many tests. It will be huge time saver in the future.

Please compare the following snipped with current test code. 

```typescript
if("User can see apis on the page", async () => {
  var backendData = Banked.createApi(TestApis.Create());
  
  const portal: DeveloperPortal = await DeveloperPortal.create(browser);
  const apisPage: ApisPage = await portal.openApisPage();
  const apisWidget: ApisWidget = await apisPage.getApisWidget();
  expect(await apisWidget.getListLength()).to.equal(backendData.apis.length);
  // or just asking ApisPage for most common elements/operations
  expect(await apisPage.getApiListLenght()).to.equal(backendData.apis.length);
});
```

File: .github/workflows/test.yml
Comment (f-alizada) [reply .github/workflows/test.yml:36]: Yes to run it in the background and not to wait to complete.

File: tests/e2e/runtime/apis.spec.ts
Comment (f-alizada) [reply tests/e2e/runtime/apis.spec.ts:37]: Thank you @Mielek. Totally agree with both suggestions, and the example looks great, do you mind if it will be done after this pr gets merged so we could populate by tests + get more feedback on how it is and implement the suggestions all at once as an improvement?

Comment (Mielek) [reply tests/e2e/runtime/apis.spec.ts:37]: I am ok with that, just want to make sure that it is not lost and it will be done. This is a big opportunity to have a great guideline how to properly do such tests.

**Review by Mielek — APPROVED** [review]

**Pull request merged**
